### PR TITLE
#2481: add definition of currentPlayer

### PR DESCRIPTION
**Pull request by yingfeng-iu**
Issue #2481: add definition of currentPlayer

### CODE REVIEW

**Comment by cjcolvar** [comment]
@adamjarling we are still using it in a few places the most notable being the Player API:
 https://github.com/avalonmediasystem/avalon/blob/develop/app/views/master_files/_player.html.erb#L31-L38
Is there a better way to reference the currently loaded media element player?  Should this code be pushed into the player instead of on the page?

**Comment by yingfeng-iu** [comment]
Chris, can I or someone squash the commits when merging the PR? 

Adam, thanks for the comment. currentPlayer is referenced in window.addEventListener in _player.html.erb, that's where the console error came from. In the current code, both mejs2 and mejs4 are provided as options. I guess if we don't need mejs2 player anymore, we can just use mejs4AvalonPlayer.player directly in place of currentPlayer. Would you prefer me to do that? Although I just took a quick look, looks like refresh_info.js.erb also reference currentPlayer, so we'll need to change there too. Let me know what's the preference. Having currentPlayer might have a small advantage for the future if we want to keep flexibility of having multiple options for players.

**Comment by phuongdh** [comment]
@yingfeng-iu I notice you're using 4-space tab for indentation while our code use 2-space indent.

File: app/views/master_files/_mejs4_player_js.html.erb [hunk 1]
```diff
@@ -12,4 +12,5 @@ var mejs4AvalonPlayer = new MEJSPlayer({
   features: ['playpause', 'current', 'progress', 'duration', 'volume', 'quality', 'fullscreen'],
   highlightRail: true
 });
+	currentPlayer = mejs4AvalonPlayer.player;
```
Comment (phuongdh) [inline app/views/master_files/_mejs4_player_js.html.erb]: I don't think there should be an indentation here?

File: app/views/media_objects/_mejs4_player_js.html.erb [hunk 2]
```diff
@@ -16,4 +16,5 @@ var mejs4AvalonPlayer = new MEJSPlayer({
   features: mejs4ConfigFeatures,
   highlightRail: true
 });
+	currentPlayer = mejs4AvalonPlayer.player;
```
Comment (phuongdh) [inline app/views/media_objects/_mejs4_player_js.html.erb]: Same indentation issue in these similar places

**Comment by adamjarling** [comment]
Ah, ok.  @cjcolvar In general, it'd be better to handle player events on the player instance object rather than `window`, but there might be a reason for doing so... I'm not sure how/why it was implemented.

@yingfeng-iu Yea, referencing the player via `mejs4AvalonPlayer.player` may be a way to reference the current player instance as well.  Once I get this playlist items page worked out (and a bunch of more structural changes I have to make to the main wrapper file to get it working), we should have a 10,000 foot look at our JS approach with the player, plugins and talk through how the whole thing works.  Assuming it'll work:)

**Comment by cjcolvar** [comment]
@adamjarling For the player api it is really the window that needs to be listening for the event: https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
This code is for the player embedded within an iframe to be controlled by the parent page.

**Comment by adamjarling** [comment]
@cjcolvar Got it. I keep forgetting the embed scenario.

File: app/views/master_files/_mejs4_player_js.html.erb (hunk 1 above)
Comment (yingfeng-iu) [reply app/views/master_files/_mejs4_player_js.html.erb]: removed indentations

File: app/views/media_objects/_mejs4_player_js.html.erb (hunk 2 above)
Comment (yingfeng-iu) [reply app/views/media_objects/_mejs4_player_js.html.erb]: removed indentations

Comment (yingfeng-iu) [reply app/views/media_objects/_mejs4_player_js.html.erb]: removed indentations

**Pull request merged**